### PR TITLE
Add mistral h100 truss

### PR DESCRIPTION
**Pull request by pankajroark**
The trt llm template wants to update a bunch of models, I haven't applied those changes here, so ci would fail. Will resolve that offline with @aspctu

### CODE REVIEW

**Comment by pankajroark** [comment]
Main things to review here are changes to generate.yaml and the very  minor updates to READMEs, rest is generate via trt-llm template.

File: mistral/mistral-7b-instruct-chat-trt-llm-h100/README.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
+# Mistral-7B-Instruct-Chat Truss
+
+This is a [Truss](https://truss.baseten.co/) for Mistral 7B Instruct. This README will walk you through how to deploy this Truss on Baseten to get your own instance of Mistral 7B Instruct.
+
+**Warning: This example is only intended for usage on a single A100, changing your resource type for this deployment will result in unsupported behavior**
```
Comment (joostinyi) [inline mistral/mistral-7b-instruct-chat-trt-llm-h100/README.md:5]: revise copy to account for H100

File: mistral/mistral-7b-instruct-chat-trt-llm-h100/config.yaml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  tokenizer_repository: mistralai/Mistral-7B-Instruct-v0.2
+model_name: Mistral 7B Instruct Chat TRT-LLM
+python_version: py311
+requirements:
+- tritonclient[all]
+- transformers==4.34.1
+- jinja2==3.1.3
+resources:
+  accelerator: A100
+  use_gpu: true
```
Comment (joostinyi) [inline mistral/mistral-7b-instruct-chat-trt-llm-h100/config.yaml:24]: Revise to use H100

File: mistral/mistral-7b-instruct-chat-trt-llm-h100/config.yaml [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+runtime:
+  num_workers: 2
```
Comment (joostinyi) [inline mistral/mistral-7b-instruct-chat-trt-llm-h100/config.yaml:26]: Let's make a note to drop this `num_workers` configuration from this example once https://github.com/basetenlabs/truss/pull/801 is in

File: mistral/mistral-7b-instruct-chat-trt-llm-h100/config.yaml [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+  tags:
+  - text-generation
+  - openai-compatible
+  tensor_parallelism: 1
+  tokenizer_repository: mistralai/Mistral-7B-Instruct-v0.2
+model_name: Mistral 7B Instruct Chat TRT-LLM
+python_version: py311
+requirements:
+- tritonclient[all]
+- transformers==4.34.1
```
Comment (joostinyi) [inline mistral/mistral-7b-instruct-chat-trt-llm-h100/config.yaml:20]: I've had some issues with this version of transformers with mistral tokenizer in practice with regards to some special tokens (see this other [example](https://github.com/basetenlabs/truss-examples/blob/f5f44701d62110875802e80c99fa1abe5223b08e/mistral/mistral-7b-instruct-chat-trt-llm/config.yaml#L21C3-L21C23) with pin to `4.35.0`)

**Review by joostinyi — CHANGES_REQUESTED** [review]
Requesting changes due to the resource requests. We have some inflight work from @aspctu in the truss repo that should help with this code duplication problem

File: mistral/mistral-7b-instruct-chat-trt-llm-h100/config.yaml (hunk 2 above)
Comment (pankajroark) [reply mistral/mistral-7b-instruct-chat-trt-llm-h100/config.yaml:24]: good catch!

**Comment by pankajroark** [comment]
Very useful review, all comments should be taken care of now.

**Review by joostinyi — APPROVED** [review]
LGTM

**Pull request merged**